### DISCUSSION TITLE
Events list in documentation?

### DISCUSSION THREAD

**Discussion by Woloja** [uNmAnNeR/imaskjs#520] [Q&A] [answered]
Hey!
I am trying to find in the documentation a list of available events that can be handled. Unfortunately found only 2 (accept, complete) in the guide. Or many are not available?
![image](https://user-images.githubusercontent.com/48343986/126673756-888c3970-956d-4dfe-8fca-332f53043ae1.png)

**Comment by uNmAnNeR** [comment, marked as answer]
hi! yep, that's it. is not enough?

**Reply by Woloja** [reply]
Ohh, okey)))
I was hoping there is more.  In my case, I needed an input error event when the mask blocks a character.

**Reply by uNmAnNeR** [reply]
this becomes a bit complicated. IMask can handle chars sequences as well as single char inputs. For example you can paste a string in mask and some chars are accepted and some are declined. Single accept event solves the problem. But that could be an issue if we want to have let say `decline` event. When should it be emitted? Or should we emit both (I am not going to go this way). So I don't see clear solution for the case that's why decline event still not exist.

**Reply by Woloja** [reply]
Thanks I got it.
I already solved my problem with an ordinary listener.  In my case, I just need to notify the user that he is entering something "bad" 🤣

**Reply by uNmAnNeR** [reply]
yep, exactly. native `input`+`accept` can help

**Reply by laptopmutia** [reply]
I try to make the listener but  the console log returning this  `Uncaught TypeError: Cannot read properties of undefined (reading 'superMark')`

I'm trying to use it in stimulus controller
```javascript
import { Controller } from "@hotwired/stimulus"
import IMask from "imask"

export default class extends Controller {
  static targets = ["debitSource", "debitTarget", "creditSource", "creditTarget"]
  superMark
  connect() {
    this.superMark = IMask(this.debitSourceTarget, {
      mask: Number,  // enable number mask

      // other options are optional with defaults below
      scale: 2,  // digits after point, 0 for integers
      signed: true,  // disallow negative
      thousandsSeparator: '.',  // any single char
      padFractionalZeros: true,  // if true, then pads zeros at end to the length of scale
      normalizeZeros: false,  // appends or removes zeros at ends
      radix: ',',  // fractional delimiter
      mapToRadix: [','],  // symbols to process as radix

      // additional number interval options (e.g.)
      min: 0,
    });

    this.superMark.on("accept", function () { console.log(this.superMark.value) });
  }
}
```
